csharp/DateTimeNanos.cs

```diff
@@ -89,18 +137,17 @@ public string ToString(string? format, IFormatProvider? formatProvider = null)
             return DateTime.ToString(adjustedFormat, formatProvider);
         }
 
-        /// <summary>
-        /// Minimum DateTime representable: 1677-09-21 00:12:43.
-        /// </summary>
-        public static readonly DateTime MinDateTimeValue = new DateTimeNanos(long.MinValue).DateTime;
-
-        /// <summary>
-        /// Maximum DateTime representable: 2262-04-11 23:47:16.
-        /// </summary>
-        public static readonly DateTime MaxDateTimeValue = new DateTimeNanos(long.MaxValue).DateTime;
-
-        private const long DateTimeOffset = 621355968000000000; // new DateTime(1970, 01, 01).Ticks
-
+        private const long DateTimeOffsetTicks = 621355968000000000; // new DateTime(1970, 01, 01).Ticks
         private const string DefaultFormat = "yyyy-MM-dd HH:mm:ss.fffffffff";
+ 
+        private static long DotnetTicksToNanosSinceEpoch(long dotnetTicks)
+        {
+            return (dotnetTicks - DateTimeOffsetTicks) * 1_000_000L / TimeSpan.TicksPerMillisecond;
```

Comment:
I was about to say you could change `private static readonly long NanosPerTick` to `private const long NanosPerTick` to fix the issue with it being zero when evaluating `MinDateTimeValue` and `MaxDateTimeValue`, which might be cleaner? If you want to expand it out here and below it would be good to keep the operation order the same as it was before to avoid potential overflow:
```suggestion
            return (dotnetTicks - DateTimeOffsetTicks) * (1_000_000L / TimeSpan.TicksPerMillisecond);
```

Reply:
Interesting - didn't know the compiler distinguished between constant and static readonly like that, thought I'd just made a silly mistake.

Reply:
Yeah I didn't realise what had caused the test failures initially either, it makes sense now but didn't occur to me that the static member order would be important.